discovery/roles/configure_xcat/common/tasks/create_osimage.yml

```diff
@@ -62,7 +86,11 @@
         path: "{{ tmp_dir }}"
         state: absent
 
-    - name: Get osimage name
+    - name: Set osimage pkgdir
+      ansible.builtin.set_fact:
+        xcat_install_dir: "{{ xcat_install_dir }}/{{ provision_os }}{{ provision_os_version }}/{{ iso_architecture }}"
```

Comment:
provision_os is not the correct way to do it. rhels9.4.0 is the format. we have to osname for osimage itself